### ISSUE TITLE
Still getting System.TypeLoadException with generics and C# covariant return types

### ISSUE THREAD

**Issue by adamjones1** [dotnet/runtime#47007]
This is a continuation of https://github.com/dotnet/runtime/issues/45344 (which has had discussion closed). The issue is still persisting for me on 5.0.102.

Repro:

```
_ = new Derived<string, int>();

class Base<T1>
{
    public virtual Base<T1> Method() => null;
}
class Derived<T1, T2> : Base<T1>
{
    public override Derived<T1, T2> Method() => null;
}
```

Note that this does work if doing any other combination of `string`/`int` as type arguments into `Derived<,>`.

EDIT: Apologies, I must have been looking at another issue when I thought the original was closed off - let me know if I should copy this into it instead and let this one get shut.

**Comment by hoyosjs** [comment]
@janvorli

**Comment by janvorli** [comment]
Ok, so this is a new failure that is still not fixed. I've checked with the latest master branch state and it fails.

**Comment by adamjones1** [comment]
Hi, this doesn't appear to be fixed with 5.0.103? Is there a timeline for the release of this fix? I can still repro with the same code sample.

**Comment by janvorli** [comment]
@adamjones1 the fix didn't make it to 5.0.3 runtime by a couple of days, so it will be in the next update (we release updates with monthly cadence).

**Comment by adamjones1** [comment]
I've just seen this still isn't fixed in the sdk 5.0.200 out this week - do you know when the fix will be released?

**Comment by hoyosjs** [comment]
5.0.200 still contains runtime 5.0.3 according to https://github.com/dotnet/core/blob/main/release-notes/5.0/releases.json

5.0.4 builds contain this fix, and I've made sure of that. As for when it's going to be released, I can't say the date, but it would be in the next release. So I'd expect this to happen within a month.

cc: @rbhanda
